kv/dist_sender.go

```diff
@@ -44,9 +44,13 @@ const (
 	retryBackoff           = 1 * time.Second
 	maxRetryBackoff        = 30 * time.Second
 
-	// Maximum number of ranges to return from an internal range lookup.
-	// TODO(mrtracy): This value should be configurable.
-	rangeLookupMaxRanges = 8
+	// The default maximum number of ranges to return
+	// from an internal range lookup.
+	defaultRangeLookupMaxRanges = 8
+	// The default size of the leader cache.
+	defaultLeaderCacheSize = 1 << 20
```

Comment:
1M entries seems high. I think a cache like this will work fine with 1 << 6 entries.